### PR TITLE
fix update project model name

### PR DESCRIPTION
**Pull request by blanchco**
# Description

* Fix to update the project model name after editing. we do the same logic in the `get` so ive added it to the `update`

### CODE REVIEW

File: packages/server/src/main/java/software/uncharted/terarium/hmiserver/service/data/ProjectAssetService.java [hunk 1]
```diff
@@ -107,6 +107,9 @@ public void updateByAsset(final TerariumAsset asset, final Schema.Permission has
 		if (!projectAssets.isEmpty()) {
 			projectAssets.forEach(projectAsset -> {
 				projectAsset.setAssetName(asset.getName());
+				if (asset instanceof Model) {
+					projectAsset.setAssetName(((Model) asset).getHeader().getName());
+				}
```
[the remark this answers is not in the document]
Comment (YohannParis) [reply packages/server/src/main/java/software/uncharted/terarium/hmiserver/service/data/ProjectAssetService.java:112]: @dvince2 could we overwrite the `setName` of `Model.java` to write `Model.ModelHeader.name` at the same time as `Model.name`?

[the remark this answers is not in the document]
Comment (mwdchang) [reply packages/server/src/main/java/software/uncharted/terarium/hmiserver/service/data/ProjectAssetService.java:112]: I almost feel like we should manage the editable parts (name, description) separately, and not within the asset (model, dataset ... etc).

[the remark this answers is not in the document]
Comment (YohannParis) [reply packages/server/src/main/java/software/uncharted/terarium/hmiserver/service/data/ProjectAssetService.java:112]: I do not follow your resoning?

**Review by mwdchang — APPROVED** [review]

File: packages/server/src/main/java/software/uncharted/terarium/hmiserver/service/data/ProjectAssetService.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (dvince2) [reply packages/server/src/main/java/software/uncharted/terarium/hmiserver/service/data/ProjectAssetService.java:112]: Models already have a lot of custom logic happening inside of `ModelService.java`. If we need to have a custom `updateByAsset` for models it should be done in `ModelService` and override the other method (for example see `ModelService#updateAsset`.

As an aside, why is `projectAssetRepository.findByAssetId` returning a list? We're asking for an asset by ID... there should only be one asset returned ever.

**Review by dvince2 — CHANGES_REQUESTED** [review]

File: packages/server/src/main/java/software/uncharted/terarium/hmiserver/service/data/ProjectAssetService.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (blanchco) [reply packages/server/src/main/java/software/uncharted/terarium/hmiserver/service/data/ProjectAssetService.java]: @dvince2 Setting the name now done on `ModelService.java` as suggested!

**Review by dvince2 — APPROVED** [review]

**Pull request merged**